### PR TITLE
Recovery wallet ui tests

### PR DESCRIPTION
**Pull request by Daniel-Wang**
## Priority
Normal

## Description
* Add recovery tests for the following test cases:
  - I'm able to recover a wallet using a 12 word mnemonic
  - I'm able to recover a wallet using a 12 word mnemonic and passcode
  - I'm able to recover a wallet using a 24 word mnemonic
  - I'm able to recover a wallet using a 24 word mnemonic and passcode
  - I'm able to recover a wallet using a 56 character secret seed

### CODE REVIEW

File: app/src/androidTest/java/com/blockeq/stellarwallet/LaunchPage.kt [hunk 1]
```diff
@@ -48,4 +48,16 @@ object LaunchPage : BasePage() {
                 .check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
                 .perform(ViewActions.click())
     }
+
+    fun clickRecoverFromPhrase() {
```
Comment (dabitdev) [inline app/src/androidTest/java/com/blockeq/stellarwallet/LaunchPage.kt]: any methods in a class that implements POM must return the page itself.
`return this`

File: app/src/androidTest/java/com/blockeq/stellarwallet/WalletRecoveryTest.kt [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ * Instrumented test, which will execute on an Android device.
+ *
+ * IMPORTANT: make sure that the app and previous tests are not installed in the device.
+ * Run sh uninstallApk.sh first. This could be solved using orchestrator tests.
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+
+@RunWith(AndroidJUnit4::class)
+class WalletRecoveryTest {
```
Comment (dabitdev) [inline app/src/androidTest/java/com/blockeq/stellarwallet/WalletRecoveryTest.kt]: In general the test are good but you are not verifying if the recovered account is the expected wallet (secret seed or account id) during mnemonic recovery and mnemonic24 + passphrase.

Another good test is to check the if the mnemonic produced under settings is the one expected.
Same for passphrase.

Comment (Daniel-Wang) [reply app/src/androidTest/java/com/blockeq/stellarwallet/WalletRecoveryTest.kt]: Okay that's true, I'll add checks for the account id

File: app/src/androidTest/java/com/blockeq/stellarwallet/WalletRecoveryTest.kt [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        clearWallet()
+    }
+
+
+    @Test
+    fun testRecoverWalletOption12WordsWithPassphrase() {
+        LaunchPage.onPageLoaded().clickRecoverFromPhrase()
+        RecoveryWalletPage.onPageLoaded().putPhrase(mnemonic12)
+        RecoveryWalletPage.onPageLoaded().clickPassphrase()
+        RecoveryWalletPage.onPageLoaded().proceedWithPassphrase(activityTestRule.activity.applicationContext, passphrase)
```
Comment (dabitdev) [inline app/src/androidTest/java/com/blockeq/stellarwallet/WalletRecoveryTest.kt]: keep the applicationContext at the top as a member. You could initialize `onBefore()` method. Could be is cleaner to get the context calling `InstrumentationRegistry.getTargetContext();`

File: app/src/androidTest/java/com/blockeq/stellarwallet/ReceivePage.kt [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import android.support.test.espresso.matcher.ViewMatchers
+import android.support.test.espresso.matcher.ViewMatchers.withText
+
+object ReceivePage : BasePage() {
+    override fun onPageLoaded(): ReceivePage {
+        onView(ViewMatchers.withId(R.id.addressEditText))
+        return this
+    }
+
+    fun checkAccount(accountId : String): ReceivePage {
```
Comment (dabitdev) [inline app/src/androidTest/java/com/blockeq/stellarwallet/ReceivePage.kt]: how about changing `checkAccount` to `assertAccount`?

**Pull request merged**